src/_pytest/main.py

```diff
@@ -293,7 +293,9 @@ def pytest_ignore_collect(path, config):
     if excludeopt:
         ignore_paths.extend([py.path.local(x) for x in excludeopt])
 
-    if py.path.local(path) in ignore_paths:
+    if any(
+        py.path.local(path).fnmatch(str(ignore_path)) for ignore_path in ignore_paths
```

Comment:
Perhaps we should use `fnmatch.fnmatch(six.text_type(path))` instead? IIRC `py.path`'s `fnmatch` differs from `fnmatch.fnmatch` slightly, and given that we want to eventually get rid of `py.path`, we should avoid depending on its semantics further.